omnigibson/action_primitives/starter_semantic_action_primitives.py

```diff
@@ -0,0 +1,1898 @@
+        """
+        eef_pos = eef_pose[0]
+        eef_ori = T.quat2axisangle(eef_pose[1])
+        end_conf = np.append(eef_pos, eef_ori)
+
+        with UndoableContext(self.robot, self.robot_copy, "original") as context:
+            plan = plan_arm_motion_ik(
+                robot=self.robot,
+                end_conf=end_conf,
+                context=context,
+                torso_fixed=TORSO_FIXED,
+            )
+
+        # plan = self._add_linearly_interpolated_waypoints(plan, 0.1)
+        if plan is None:
+            raise ActionPrimitiveError(
+                ActionPrimitiveError.Reason.PLANNING_ERROR,
+                "There is no accessible path from where you are to the desired joint position. Try again"
+            )
+        
+        # Follow the plan to navigate.
+        indented_print("Plan has %d steps", len(plan))
+        for i, target_pose in enumerate(plan):
+            target_pos = target_pose[:3]
+            target_quat = T.axisangle2quat(target_pose[3:])
+            indented_print("Executing grasp plan step %d/%d", i + 1, len(plan))
+            yield from self._move_hand_direct_ik((target_pos, target_quat), ignore_failure=True, in_world_frame=False, stop_if_stuck=stop_if_stuck)
+
+    def _add_linearly_interpolated_waypoints(self, plan, max_inter_dist):
+        """
+        Adds waypoints to the plan so the distance between values in the plan never exceeds the max_inter_dist.
+
+        Args:
+            plan (Array of arrays): Planned path
+            max_inter_dist (float): Maximum distance between values in the plan
+        
+        Returns:
+            Array of arrays: Planned path with additional waypoints
+        """
+        plan = np.array(plan)
+        interpolated_plan = []
+        for i in range(len(plan) - 1):
+            max_diff = max(plan[i+1] - plan[i])
+            num_intervals = ceil(max_diff / max_inter_dist)
+            interpolated_plan += np.linspace(plan[i], plan[i+1], num_intervals, endpoint=False).tolist()
+        interpolated_plan.append(plan[-1].tolist())
+        return interpolated_plan
+    
+    def _compute_delta_command(self, controller_name, diff_joint_pos, gain=1.0, min_action=0.0):
+        controller = self.robot._controllers[controller_name]
+        control_limits = controller._control_limits[controller.control_type]
+        gain = 1.0
+        min_action = 0.0
+        # for joints not within thresh, set a minimum action value
+        # for joints within thres, set action to zero
+        delta_action = gain * diff_joint_pos
+        delta_action[abs(delta_action) < min_action] = np.sign(delta_action[abs(delta_action) < min_action]) * min_action
+        delta_action[abs(diff_joint_pos) < JOINT_POS_DIFF_THRESHOLD] = 0.0
+        # TODO - this is temporary
+        if TORSO_FIXED:
+            delta_action = np.concatenate([np.zeros(1), delta_action])
+        
+        return delta_action
```

Comment:
The goal is to make sure that if we are outputting an action, it's never too small. It's an attempt to avoid diminishing input as you approach the goal. May cause oscillation however so we'll investigate this later too.